File: tests/e2e/fixtures/code/sample.scala
// Scala: case classes, pattern matching, for-comprehension.
package nfe

import scala.io.Source

case class Note(path: String, tags: List[String] = Nil) {
  def size: Long = Source.fromFile(path).length
}

object Scanner {
  val Limit = 5L * 1024 * 1024

  def groupByTag(notes: Seq[Note]): Map[String, Seq[Note]] =
    (for { n <- notes; t <- n.tags } yield t -> n).groupMap(_._1)(_._2)

  def describe(n: Int): String = n match {
    case 0 => "no tags"
    case x if x > 100 => s"many tags: $x"
    case x => s"$x tags"
  }
}

File: tests/e2e/fixtures/code/sample.erl
%% Erlang: modules, records, pattern matching, list comprehensions.
-module(scanner).
-export([group_by_tag/1, describe/1]).

-define(LIMIT, 5 * 1024 * 1024).
-record(note, {path, tags = [], size = 0}).

group_by_tag(Notes) ->
    Pairs = [{T, N} || N = #note{tags = Tags, size = S} <- Notes, S =< ?LIMIT, T <- Tags],
    lists:foldl(fun({T, N}, Acc) -> maps:update_with(T, fun(L) -> [N | L] end, [N], Acc) end,
                #{}, Pairs).

describe(0) -> "no tags";
describe(N) when N > 100 -> io_lib:format("many tags: ~p", [N]);
describe(N) -> io_lib:format("~p tags", [N]).
